template/description.ext

```diff
@@ -0,0 +1,46 @@
+#include "script_component.hpp"
+#include "gui\gui_init.hpp"
+
+/* Respawn stuff */
+respawn = 3;
+respawnDelay = 1e10;
+respawnTemplates[] = {"afm_respawn_default", "Counter"};
+// If mission doesn't feature standard respawn timer, "Counter" option should be removed.
+// This applies to permadeath missions and when using custom respawn based on events or whatever.
+// For compatibility with spectator module:
+respawnTemplatesVirtual[] = {"Spectator"};
```

Comment:
I'm not sure whether `respawnTemplatesVirtual` is actually useful. I remember talking with someone about this, but I don't remember the outcome.

Reply:
It's useless, "Spectator" makes you go into spectator upon dying, this makes no sense for virtual units. VirtualSpectator logic makes you go into spectator by itself.